src/values/AbstractObjectValue.js

```diff
@@ -482,16 +482,16 @@ export default class AbstractObjectValue extends AbstractValue {
 
   $GetPartial(P: AbstractValue | PropertyKeyValue, Receiver: Value): Value {
     if (!(P instanceof AbstractValue)) return this.$Get(P, Receiver);
-    invariant(this === Receiver, "TODO #1021");
     if (this.values.isTop()) {
-      if (this.isSimpleObject() && this.isIntrinsic()) {
+      if ((this.isSimpleObject() && this.isIntrinsic()) || this.$Realm.isInPureScope()) {
```

Comment:
The simple object check is to ensure this can’t have side effects. If no such guarantee exists we need to havoc this object, the receiver and the property value.